jsdependencies-sbt-plugin/src/main/scala/org/scalajs/jsdependencies/sbtplugin/JSDependenciesPlugin.scala

```diff
@@ -0,0 +1,364 @@
+package org.scalajs.jsdependencies.sbtplugin
+
+import scala.collection.mutable
+import scala.util.Try
+
+import java.io.InputStreamReader
+
+import sbt._
+import sbt.Keys._
+
+import org.scalajs.core.ir.Utils.escapeJS
+
+import org.scalajs.core.tools.io.{IO => toolsIO, _}
+import org.scalajs.core.tools.json._
+
+import org.scalajs.jsenv.VirtualFileMaterializer
+
+import org.scalajs.sbtplugin.ScalaJSPlugin
+import org.scalajs.sbtplugin.ScalaJSPlugin.autoImport._
+import org.scalajs.sbtplugin.ScalaJSPluginInternal.scalaJSSourceFiles
+
+import org.scalajs.jsdependencies.core._
+import org.scalajs.jsdependencies.core.DependencyResolver.DependencyFilter
+import org.scalajs.jsdependencies.core.ManifestFilters.ManifestFilter
+
+object JSDependenciesPlugin extends AutoPlugin {
+  override def requires: Plugins = ScalaJSPlugin
+
+  object autoImport {
+    import KeyRanks._
+
+    val scalaJSNativeLibraries = TaskKey[Attributed[Seq[VirtualJSFile with RelativeVirtualFile]]](
+        "scalaJSNativeLibraries", "All the *.js files on the classpath", CTask)
+
+    val packageJSDependencies = TaskKey[File]("packageJSDependencies",
+        "Packages all dependencies of the preLink classpath in a single file.", AMinusTask)
+
+    val packageMinifiedJSDependencies = TaskKey[File]("packageMinifiedJSDependencies",
+        "Packages minified version (if available) of dependencies of the preLink " +
+        "classpath in a single file.", AMinusTask)
+
+    val jsDependencyManifest = TaskKey[File]("jsDependencyManifest",
+        "Writes the JS_DEPENDENCIES file.", DTask)
+
+    val jsDependencyManifests = TaskKey[Attributed[Traversable[JSDependencyManifest]]](
+        "jsDependencyManifests", "All the JS_DEPENDENCIES on the classpath", DTask)
+
+    val requiresDOM = SettingKey[Boolean]("requiresDOM",
+        "Whether this projects needs the DOM. Overrides anything inherited through dependencies.", AMinusSetting)
+
+    val jsDependencies = SettingKey[Seq[AbstractJSDep]]("jsDependencies",
+        "JavaScript libraries this project depends upon. Also used to depend on the DOM.", APlusSetting)
+
+    val jsDependencyFilter = SettingKey[DependencyFilter]("jsDependencyFilter",
+        "The filter applied to the raw JavaScript dependencies before execution", CSetting)
+
+    val jsManifestFilter = SettingKey[ManifestFilter]("jsManifestFilter",
+        "The filter applied to JS dependency manifests before resolution", CSetting)
+
+    val resolvedJSDependencies = TaskKey[Attributed[Seq[ResolvedJSDependency]]]("resolvedJSDependencies",
+        "JS dependencies after resolution.", DTask)
+
+    /** Internal task to calculate whether a project requests the DOM
+     *  (through jsDependencies or requiresDOM) */
+    val scalaJSRequestsDOM = TaskKey[Boolean]("scalaJSRequestsDOM",
+        "Scala.js internal: Whether a project really wants the DOM. " +
+        "Calculated using requiresDOM and jsDependencies", KeyRanks.Invisible)
```

Comment:
This setting is unused. We used to use it to determine the `JSEnv` automatically. I think it can just be dropped (OK to do later, if you want clean deprecation).